lib/src/model/canonicalization.dart

```diff
@@ -74,10 +74,56 @@ final class _Canonicalization {
 
   _Canonicalization(this._element);
 
+  /// Append an encoded form of the given [component] to the given [buffer].
+  void _encode(StringBuffer buffer, String component) {
+    var length = component.length;
+    for (var i = 0; i < length; i++) {
+      var currentChar = component.codeUnitAt(i);
+      if (currentChar == _separatorChar) {
+        buffer.writeCharCode(_separatorChar);
+      }
+      buffer.writeCharCode(currentChar);
+    }
+  }
+
+  String _getElementLocation(Element2 element) {
```

Comment:
Could we add a comment about where this comes from, so that we can look to the other implementation if bugs arise?

Reply:
added